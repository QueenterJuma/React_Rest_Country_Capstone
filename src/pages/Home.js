import earth from '../image/earth.jpg';
import Input from '../components/Input';
import HomeInfo from '../components/Homeinfo';

const Home = () => (
  <div>
    <div className="content1">
      <div className="img-box">
        <img className="world-img spiner" src={earth} alt="world" />
      </div>
      <div className="banner-text">
        <p className="ban1"> The Continents</p>
      </div>
    </div>
    <div>
      <Input />
    </div>
    <h2 className="stats">STATS BY COUNTRY</h2>
    <HomeInfo />
  </div>
);

export default Home;
