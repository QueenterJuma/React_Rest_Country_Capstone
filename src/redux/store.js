import { configureStore } from '@reduxjs/toolkit';
import countriesReducer from './country/CountryInfoSlice';

const store = configureStore({
  reducer: {
    country: countriesReducer,
  },
});

export default store;
