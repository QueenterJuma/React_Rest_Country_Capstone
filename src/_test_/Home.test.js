import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Home from '../pages/Home';
import store from '../redux/store';

describe('Home', () => {
  test('renders correctly', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Home />
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
