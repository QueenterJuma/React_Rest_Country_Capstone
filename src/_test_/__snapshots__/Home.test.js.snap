// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div>
  <div
    className="content1"
  >
    <div
      className="img-box"
    >
      <img
        alt="world"
        className="world-img spiner"
        src="earth.jpg"
      />
    </div>
    <div
      className="banner-text"
    >
      <p
        className="ban1"
      >
         The Continents
      </p>
    </div>
  </div>
  <div>
    <div
      className="input-container"
    >
      <input
        className="search-input"
        onChange={[Function]}
        placeholder="Search"
        type="text"
        value=""
      />
    </div>
  </div>
  <h2
    className="stats"
  >
    STATS BY COUNTRY
  </h2>
  <div
    className="homeinfo-container"
  />
</div>
`;

exports[`renders correctly 1`] = `
<div>
  <div
    className="content1"
  >
    <div
      className="img-box"
    >
      <img
        alt="world"
        className="world-img spiner"
        src="earth.jpg"
      />
    </div>
    <div
      className="banner-text"
    >
      <p
        className="ban1"
      >
         The Continents
      </p>
    </div>
  </div>
  <div>
    <div
      className="input-container"
    >
      <input
        className="search-input"
        onChange={[Function]}
        placeholder="Search"
        type="text"
        value=""
      />
    </div>
  </div>
  <h2
    className="stats"
  >
    STATS BY COUNTRY
  </h2>
  <div
    className="homeinfo-container"
  />
</div>
`;
